src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -496,6 +497,15 @@ public override Type[] GetOptionalCustomModifiers()
                 m_signature.GetCustomModifiers(PositionImpl + 1, false);
         }
 
+        public override bool Equals(object? obj) =>
```

Comment:
Do we really need to fix ParameterInfo? 

It is unlikely that ParameterInfo is used as a key anywhere. We are not maintaining strong referential equality for ParameterInfo today. For example, the intialization at https://github.com/dotnet/runtime/blob/0ba3d3124e1ba3bc5be676ca33eaca0868858dd5/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.CoreCLR.cs#L80 is not thread safe - two threads may get two different instances of ParameterInfo.

Reply:
not sure, just trying to cover all `*Infos`, will revert then